### ISSUE TITLE
Video isn't using the whole screen

### ISSUE THREAD

**Issue by loganmc10** [mupen64plus-ae/mupen64plus-ae#613]
This is on a Shield Tablet, I have "immersive mode" selected and I tried Glide64 and GLideN64 (crop mode auto). Am I doing something wrong? Or is there some way for me to look into this? These games use the whole screen in mupen64plus on Linux

![screenshot_20170303-142758](https://cloud.githubusercontent.com/assets/848146/23569729/1a6f6b36-001e-11e7-93c4-12c727afc29a.png)

**Comment by loganmc10** [comment]
To be clear, I'm talking about the black bars on the top and bottom, not the sides

**Comment by fzurita** [comment]
Strange, they use the full screen in my device. Does the zoom option work for you? It's a per game setting, so select the game and go to settings, and increase the zoom level there.

**Comment by loganmc10** [comment]
Ya the zoom setting makes it fill the screen

**Comment by fzurita** [comment]
I don't understand though, it should work without that. Do other games have the same issue?

**Comment by loganmc10** [comment]
Every game has the same issue. Additionally, setting screen scaling to "stretch" instead of "original", stretches the image all the way left and right, but the bars on the top and bottom remain

**Comment by fzurita** [comment]
What if you changes the display resolution?

**Comment by loganmc10** [comment]
Everything from 320x240 all the way to native makes no difference.

I'll get Android Studio setup in the next few days and see if I can figure out what might make a difference

**Comment by fzurita** [comment]
Sure, if you really want. I'll try to look at our code to see what's up. There relevant code is in GameActivity.java in the onCreate method. There is nothing that complicated in there.

**Comment by loganmc10** [comment]
These are screenshots from my phone (Xperia XA). Mario 64 still has the bars but they are small, Zelda has a very slight bar, I have the menu open to show the difference

![screenshot_20170303-170159](https://cloud.githubusercontent.com/assets/848146/23573543/d579d08c-0033-11e7-879e-15096c9298d9.png)

![screenshot_20170303-170030](https://cloud.githubusercontent.com/assets/848146/23573546/e06b91e2-0033-11e7-92ce-61103398453c.png)

**Comment by loganmc10** [comment]
Maybe Mario 64 just has the bars, I can't remember for sure right now what it's like on Linux, although I thought auto crop would fix that. On my phone most games look like Zelda, almost no bar

**Comment by loganmc10** [comment]
Ok Mario has those bars on Linux even with auto crop and Zelda has that slight bar as well, so that's all normal (so my phone is normal), its just my Shield Tablet that seems to have this issue

**Comment by fzurita** [comment]
Yeah, it's probably related to resolution and DPI.

**Comment by fzurita** [comment]
Most likely we have the resolution calculation wrong somewhere

**Comment by fzurita** [comment]
If you are really interested, take a look at GlobalPrefs.java. See how we calculate ```videoSurfaceHeightOriginal```

I have the feeling that we are not getting correct values for the shield tablet.

**Comment by loganmc10** [comment]
So I realized the problem is with "immersive mode". See the screenshots below, the first is with immersive mode off, and it looks correct, the second is with it on.

It looks like it's not taking into account the fact that the "Android bar" (I don't know it's proper name) is gone

![screenshot_20170304-131236](https://cloud.githubusercontent.com/assets/848146/23581950/7f17d938-00dc-11e7-98cb-7e1e1e1d6383.png)

![screenshot_20170304-131300](https://cloud.githubusercontent.com/assets/848146/23581954/85f672a0-00dc-11e7-8268-09171ed784ec.png)

**Comment by fzurita** [comment]
If you can get it to compile, can you tell me what you are getting for videoSurfaceWidthOriginal and videoSurfaceHeightOriginal in GlobalPrefs?

**Comment by fzurita** [comment]
I checked in a fix that could **potentially** fix the issue. If you can build the app, you can try it.

**Comment by loganmc10** [comment]
I tried the latest master and it still does the same thing, I'll play around with it before, shouldn't it be getRealMetrics instead of getMetrics for immersive mode?

**Comment by fzurita** [comment]
Maybe, if you want to try it. It's hard to tell what is going to work for all devices.

**Comment by loganmc10** [comment]
Ya I'll figure this all out tomorrow and let you know

**Comment by loganmc10** [comment]
So I don't think there is anything wrong with the calculations. It doesn't seem to be getting the right value for "isImmersiveModeEnabled".

If I use "getRealMetrics" (or getRealSize), and if I force isImmersiveModeEnabled to true, then it works. I also tested this in the Android Emulator (tablet) and I found the same thing.

This probably doesn't show up as an issue on the phone because the "Android bar" is on the side, not the bottom.

**Comment by fzurita** [comment]
Hmmm, interesting. That's a big clue.

**Comment by fzurita** [comment]
I see the problem... we are setting that variable AFTER we use it... whoops. Looks like a mistake on my part.

**Comment by fzurita** [comment]
Ok, fixed: 70cf137f20a9bbf8da4b1bf41b2f7e97fcf70302

**Comment by loganmc10** [comment]
awesome thanks!

**Comment by LDCanetti** [comment]
I'm having the same trouble with my Xiaomi redmi note 7
![Screenshot_2020-04-03-03-50-24-199_paulscode android mupen64plus free](https://user-images.githubusercontent.com/50004997/78337327-d4747900-755e-11ea-8d51-aab1e55a7af9.jpg)

**Comment by fzurita** [comment]
There problem mentioned here was fixed a long time ago, unless we introduced a regression somewhere.

First, which version of the app are in you using. Make sure you are using the latest here:

https://play.google.com/store/apps/details?id=org.mupen64plusae.v3.fzurita

**Comment by Slipyx** [comment]
I've noticed this issue still in effect when using an external display on a chromebook. I'm using the latest mupen64plus-fz 3.0.276(beta)-free from the play store on an intel x86_64 chromebook. I've tested both glide plugins at various resolutions. It only fails to fill the whole screen when immersive mode is turned on. This only happens on the external 1080p display, using the internal 1366x768 display everything works as intended.

With immersive mode turned on:
![m64gn64-imm](https://user-images.githubusercontent.com/694193/109411203-6010a880-796e-11eb-83b0-6a4e896bd5e4.jpg)

With immersive mode off:
![m64gn64-noimm](https://user-images.githubusercontent.com/694193/109411222-846c8500-796e-11eb-97c8-ee9b4da3545c.jpg)
